File: SVR.R
#sVR
dataset= read.csv('Position_Salaries.csv')
dataset=dataset[2:3]

#library(caTools)
#set.seed(123)
#split=sample.split(dataset$Profit,SplitRatio=0.8)
#training_set=subset(dataset,split==TRUE)
#test_set=subset(dataset,split==FALSE)



#fitting SVR to the dataset

regressor =svm(formula=Salary~.,
           data=dataset,
           type='eps-regression')

#prediction with polynmial reg model

y_pred=predict(regressor,data.frame(Level=6.5))

#visualising the regression modellibrary(ggplot2)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),color='red')+
  geom_line(aes(x=dataset$Level,y=predict(regressor,newdata = dataset)),color='blue')+
  ggtitle('truth or bluff(SVR model)')+
  xlab('Level')+
  ylab('salary')


#visualising the regression modellibrary(ggplot2) with smooth curves
x_grid=seq(min(dataset$Level),max(dataset$Level),0.1)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),color='red')+
  geom_line(aes(x=x_grid,y=predict(regressor,newdata = data.frame((level=x_grid)))),color='blue')+
  ggtitle('truth or bluff(reg model)')+
  xlab('Level')+
  ylab('salary')

File: SVR.py
# -*- coding: utf-8 -*-
"""
Created on Wed Mar  8 14:23:35 2017

@author: Habiba
"""

# -*- coding: utf-8 -*-
"""
Created on Tue Mar  7 22:46:42 2017

@author: Habiba
"""

# -*- coding: utf-8 -*-
"""
Created on Tue Mar  7 13:40:42 2017

@author: Habiba
"""
#SVR
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


dataset=pd.read_csv('Position_Salaries.csv')
X=dataset.iloc[:,1:2].values
y=dataset.iloc[:,2].values              

"""#splitting dataset into training and data set
from sklearn.cross_validation import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=0)"""

#feature scaling
from sklearn.preprocessing import StandardScaler
sc_X=StandardScaler()
sc_y=StandardScaler()
X=sc_X.fit_transform(X)
y=sc_y.fit_transform(y)

#fitting  SVR to data set
from sklearn.svm import SVR
regressor=SVR(kernel='rbf')
regressor.fit(X,y)

#predicting the result with polynomial regresion
y_pred=sc_y.inverse_transform(regressor.predict(sc_X.transform(np.array([[6.5]]))))

#visualise the SVR results
plt.scatter(X,y,color='red')
plt.plot(X,regressor.predict(X),color='blue')
plt.title('regression graph(SVR)')
plt.xlabel('position level')
plt.ylabel('salary')
plt.show()
